### PR TITLE
Update to U-Boot 2020.01

### PR DESCRIPTION
**Pull request by oleorhagen**
This is the latest attempt at migrating.

What is missing is changing the U-Boot environment addresses in the `defconfig` files. Only question is: 

Should we align this with the `8MiB` alignment in `Yocto`. Or should I add a commit, and have them back to `4MiB` for `mender-convert` ?

### CODE REVIEW

**Comment by ghost** [comment]
I would suggest unifying on 8MiB, just for keeping things the same everywhere, although I don't have a strong opinion. @mirzak: Do you remember if `/etc/fw_env.config` is symlinked to `/data` also in mender-convert? If so it should not affect existing devices.

Otherwise I'm not completely sure what to review here, this is a collection of preexisting patches...?

**Comment by oleorhagen** [comment]
> Otherwise I'm not completely sure what to review here, this is a collection of preexisting patches...?

Yes

**Comment by mirzak** [comment]
> Do you remember if /etc/fw_env.config is symlinked to /data also in mender-convert?

Actually, I do not think it is. https://github.com/mendersoftware/mender-convert/blob/master/configs/raspberrypi_config#L85

So changing to 8MiB is a no-go

File: include/config_mender_defines.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#define MENDER_ROOTFS_PART_A_NUMBER_HEX 2
+#define MENDER_ROOTFS_PART_B_NUMBER 3
+#define MENDER_ROOTFS_PART_B_NUMBER_HEX 3
+#define MENDER_UBOOT_STORAGE_INTERFACE "mmc"
+#define MENDER_UBOOT_STORAGE_DEVICE 0
+
+/* BB variables. */
+#define MENDER_STORAGE_DEVICE_BASE "/dev/mmcblk0p"
+#define MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_1 0x400000
+#define MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_2 0x800000
```
Comment (ghost) [inline include/config_mender_defines.h:20]: If switching back to 8MiB, then these need to be `0x800000` and `0x1000000`, respecitvely.

**Review by ghost — CHANGES_REQUESTED** [review]
The patches look ok. I guess e22d3af was just taken from one of the earlier branches?

File: include/config_mender_defines.h (hunk 1 above)
Comment (oleorhagen) [reply include/config_mender_defines.h:20]: I thought changing to `8MiB` was not possible due to the config not being on the data partition?

Comment (ghost) [reply include/config_mender_defines.h:20]: Ugh, you are absolutely right, good that you are looking out! I just got the two confused.

**Review by ghost — APPROVED** [review]

**Review by mirzak — APPROVED** [review]

**Pull request merged**